fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -4647,8 +4647,8 @@ public static class Builder implements BaseBuilder<Message, FDBRecordStore> {
         @Nonnull
         private IndexMaintainerRegistry indexMaintainerRegistry = IndexMaintainerRegistryImpl.instance();
 
-        @Nonnull
-        private IndexMaintenanceFilter indexMaintenanceFilter = IndexMaintenanceFilter.NORMAL;
+        @Nullable
+        private IndexMaintenanceFilter indexMaintenanceFilter = null;
```

Comment:
Why are we changing for a special element `NORMAL` specific to `IndexMaintenanceFilter` (that allows us to be `@Nonnull` in a lot of places, thus allowing us to call methods on it without prior checking) to a non-specific `null`?

Reply:
The requirement, as specified by @arnaud-lacurie, was to prefer the `indexMaintenanceFilter` over the new predicates iff it was explicitly set the caller (for backward compatibility, I suppose). Setting it to `NORMAL` preemptively hides the information, prevents the code below from knowing if it was explicit or not (as the user may have also set it to `NORMAL`)